antenna/configlib.py

```diff
@@ -0,0 +1,323 @@
+"""This module contains the configuration infrastructure allowing for deriving
+configuration from a .ini file and the process environment.
+
+In order of precedence:
+
+1. settings overrides (usually only applies when running tests)
+2. process environment
+3. .ini file (specified by ANTENNA_INI environment variable)
+
+Example of usage::
+
+    from antenna.configlib import config
+
+    DEBUG = config('DEBUG', default=True, parser='bool')
+
+
+This also makes it easy to do testing using the settings_override
+decorator::
+
+    from antenna.configlib import settings_override
+
+    @settings_override(FOO='bar', BAZ='bat')
+    def test_this():
+        ...
+
+
+Wait--what? Why'd write our own configuration library when there are so many
+out there?
+
+I wanted one that was flexible, supported namespaces and it needed to be easy
+to test with. That meant it needed to work with overrides and it needed to
+compose nicely. I wanted something with namespaces so that I could have two
+instances of the same component with two different configurations.
+
+"""
+
+import inspect
+import os
+import urlparse
+from ConfigParser import SafeConfigParser as ConfigParser
+from functools import wraps
+
+
+# This is a stack of overrides to be examined in reverse order
+_CONFIG_OVERRIDE = []
+
+
+# Singleton indicating a non-value
+NO_VALUE = object()
+
+
+def parse_bool(val):
+    """Parses a bool
+
+    Handles a series of values, but you should probably standardize on
+    "true" and "false".
+
+    """
+    true_vals = ('t', 'true', 'yes', 'y', '1')
+    false_vals = ('f', 'false', 'no', 'n', '0')
+
+    val = val.lower()
+    if val in true_vals:
+        return True
+    if val in false_vals:
+        return False
+
+    raise ValueError('%s is not a valid bool value' % val)
```

Comment:
I like this! That means that a typo like "not" doesn't accidentally get converted to `True` when I meant `False`.